Makefile

```diff
@@ -1,38 +1,48 @@
 GO ?= $(shell command -v go 2> /dev/null)
-DEP ?= $(shell command -v dep 2> /dev/null)
 NPM ?= $(shell command -v npm 2> /dev/null)
-HTTP ?= $(shell command -v http 2> /dev/null)
 CURL ?= $(shell command -v curl 2> /dev/null)
 MANIFEST_FILE ?= plugin.json
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+
+# You can include assets this directory into the bundle. This can be e.g. used to include profile pictures.
+ASSETS_DIR ?= assets
 
 # Verify environment, and define PLUGIN_ID, PLUGIN_VERSION, HAS_SERVER and HAS_WEBAPP as needed.
 include build/setup.mk
 
 BUNDLE_NAME ?= $(PLUGIN_ID)-$(PLUGIN_VERSION).tar.gz
 
-# all, the default target, tests, builds and bundles the plugin.
+## Checks the code style, tests, builds and bundles the plugin.
 all: check-style test dist
 
-# apply propagates the plugin id into the server/ and webapp/ folders as required.
+## Propagates plugin manifest information into the server/ and webapp/ folders as required.
 .PHONY: apply
 apply:
 	./build/bin/manifest apply
 
+## Runs govet and gofmt against all packages.
 .PHONY: check-style
-check-style: server/.depensure webapp/.npminstall gofmt govet errcheck
+check-style: webapp/node_modules gofmt govet errcheck
 	@echo Checking for style guide compliance
 
 ifneq ($(HAS_WEBAPP),)
 	cd webapp && npm run lint
 endif
 
+## Runs gofmt against all packages.
 .PHONY: gofmt
 gofmt:
 ifneq ($(HAS_SERVER),)
 	@echo Running gofmt
-	@for package in $$(go list ./server/...); do \
-		echo "Checking "$$package; \
-		files=$$(go list -f '{{range .GoFiles}}{{$$.Dir}}/{{.}} {{end}}' $$package); \
+	@cd server; \
+	initial_path=$$(pwd); \
```

Comment:
Does the code in plugin-sample not work?

Reply:
It doesn't work with submodule... But now I have packages it should work. I keep this, because I want to play with submodule (for analytics plugin and this [issue](https://github.com/manland/mattermost-plugin-analytics/issues/14))